tests/foreman/ui/test_discoveredhost.py

```diff
@@ -98,16 +100,135 @@ def test_positive_provision_pxe_host(
+
+    :CaseImportance: High
+    """
+    sat = module_discovery_sat.sat
+    provisioning_host.power_control(ensure=False)
+    mac = provisioning_host._broker_args['provisioning_nic_mac_addr']
+    wait_for(
+        lambda: sat.api.DiscoveredHost().search(query={'mac': mac}) != [],
+        timeout=1500,
+        delay=20,
+    )
+    discovered_host = sat.api.DiscoveredHost().search(query={'mac': mac})[0]
+    discovered_host.hostgroup = provisioning_hostgroup
+    discovered_host.location = provisioning_hostgroup.location[0]
+    discovered_host.organization = provisioning_hostgroup.organization[0]
+    discovered_host.build = True
+
+    discovered_host_name = discovered_host.name
+    domain_name = provisioning_hostgroup.domain.read().name
+    host_name = f'{discovered_host_name}.{domain_name}'
+
+    # Teardown
+    request.addfinalizer(lambda: sat.provisioning_cleanup(host_name))
+
+    # Change root_passwd in HostGroup, and for customization change it back during provisioning
+    root_pwd = gen_string('alpha')
+    provisioning_hostgroup.root_pass = root_pwd
+    provisioning_hostgroup.update(['root_pass'])
+    new_root_pwd = settings.server.ssh_password
+
+    @request.addfinalizer
+    def _finalize():
+        provisioning_hostgroup.root_pass = new_root_pwd
+        provisioning_hostgroup.update(['root_pass'])
+
+    # Sync and assign Ansible role to HostGroup to validate BZ:2025523
+    SELECTED_ROLE = 'theforeman.foreman_scap_client'
+    proxy_id = sat.nailgun_smart_proxy.id
+    sat.cli.Ansible.roles_sync({'role-names': f'{SELECTED_ROLE}', 'proxy-id': proxy_id})
+    result = sat.cli.HostGroup.ansible_roles_add(
+        {'name': provisioning_hostgroup.name, 'ansible-role': SELECTED_ROLE}
+    )
+    assert 'Ansible role has been associated.' in result[0]['message']
+
+    with sat.ui_session() as session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_location.name)
+        session.discoveredhosts.provision(
+            discovered_host_name,
+            provisioning_hostgroup.name,
+            module_org.name,
+            module_location.name,
+            quick=False,
+            host_values={'operating_system.root_password': new_root_pwd},
+        )
+        # Wait for provisioning to complete and report status back to Satellite
+        pending_status = 'N/A' if is_open('BZ:2258024') else 'Pending installation'
+        wait_for(
+            lambda: session.host_new.get_host_statuses(host_name)['Build']['Status']
+            != pending_status,
+            timeout=1800,
+            delay=30,
+            fail_func=session.browser.refresh,
+            silent_failure=True,
+            handle_exception=True,
+        )
+        values = session.host_new.get_host_statuses(host_name)
+        assert values['Build']['Status'] == 'Installed'
+        assert values['Execution']['Status'] == 'Last execution succeeded'
```

Comment:
Will this ensure the last execution was the ansible role one?

Reply:
yeah, it will be the last execution on host which is run immediately after provisioning